bundles/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/handler/SamsungTvHandler.java

```diff
@@ -330,6 +331,12 @@ public void thingDiscovered(DiscoveryService source, DiscoveryResult result) {
              */
             upnpThingUID = result.getThingUID();
             logger.debug("thingDiscovered, thingUID={}, discoveredUID={}", this.getThing().getUID(), upnpThingUID);
+            // We wait a little to increase the chance to have all UPnP devices being discovered before
+            // calling checkAndCreateServices
+            try {
+                Thread.sleep(2000);
+            } catch (InterruptedException e) {
+            }
```

Comment:
thingDiscovered method will be called when ever new UPnP device is discovered, so delay shouldn't be needed

Reply:
I think thingDiscovered is called only if a new OH thing is discovered.
For a new UPnP device that is ignored by the OH discovery, I don't think thingDiscovered will be called.
But I will check the code from the framework core again to be sure.

Reply:
I think 2 seconds delay is not enough as Samsung TV's can announce services very randomly, it can take minutes before all services are discovered. 

I can't remember anymore when change was done from UPnP library to DiscoveryListener and unfortunately git history is gone because of bnd build change.

Reply:
If it can take so much time, I understand my proposal makes no sense.
I propose to suppress this part from my PR.